lib/src/client.dart

```diff
@@ -246,9 +255,23 @@ class NyxxOAuth2 with ManagerMixin implements NyxxRest {
   Future<void> leaveThread(Snowflake id) => channels.leaveThread(id);
 
   @override
-  Future<List<PartialGuild>> listGuilds({Snowflake? before, Snowflake? after, int? limit}) =>
+  Future<List<UserGuild>> listGuilds({Snowflake? before, Snowflake? after, int? limit}) =>
       users.listCurrentUserGuilds(before: before, after: after, limit: limit);
 
+  Future<OAuth2Information> fetchCurrentOAuth2Information() async {
+    final route = HttpRoute()
+      ..oauth2()
+      ..add(HttpRoutePart('@me'));
+    final request = BasicRequest(route);
+    final response = await httpHandler.executeSafe(request);
+    final body = response.jsonBody as Map<String, Object?>;
+    return OAuth2Information(
+        application: PartialApplication(manager: applications, id: Snowflake.parse((body['application'] as Map<String, Object?>)['id']!)),
+        scopes: (body['scopes'] as List).cast(),
+        expiresOn: DateTime.parse(body['expires'] as String),
+        user: maybeParse(body['user'], users.parse));
+  }
+
```

Comment:
Any HTTP and paring logic should go in a manager. This could probably go in UserManager.